projects/github-actions/repo-gardening/src/tasks/flag-oss/index.js

```diff
@@ -15,7 +15,14 @@ async function flagOss( payload, octokit ) {
 	const { head, base } = pull_request;
 	const { owner, name } = repository;
 
-	if ( head.repo.full_name === base.repo.full_name ) {
+	// Check if PR author is org member
+	// https://docs.github.com/en/rest/orgs/members?apiVersion=2022-11-28#check-organization-membership-for-a-user
+	const orgMembershipRequest = await octokit.rest.orgs.checkMembershipForUser( {
+		org: owner.login,
+		username: head.user.login,
+	} );
+
+	if ( head.repo.full_name === base.repo.full_name || 204 === orgMembershipRequest.status ) {
```

Comment:
This makes sense, that would be a good addition I think.

That said, I wonder if we could save an API call in most scenarios, by keeping the 2 checks separate?

1. We would start by checking if the PR is from a branch in the same remote, like we do today. If yes we bail, like today.
2. Then, only when the PR is from a different remote we make the API call you're introducing here to figure out if the PR author is a member of the organization.

I believe it would avoid slowing things down for most of the PRs today.

What do you think?

Reply:
Good point! Applied on [703a0b0](https://github.com/Automattic/jetpack/pull/36860/commits/703a0b0e8ba092d8ffb1103909ec46c0a43db8de)